### PR TITLE
pthread_detach

### PR DESCRIPTION
**Pull request by lucasccordeiro**
Added operational model, test cases, and oracle for the `pthread_detach` function.

### CODE REVIEW

**Review by fatimahkj — APPROVED** [review]

File: src/c2goto/library/pthread_lib.c [hunk 1]
```diff
@@ -706,3 +711,26 @@ __ESBMC_HIDE:;
...
+// as detached.  When a detached thread terminates, its resources are
+// automatically released back to the system without the need for
+// another thread to join with the terminated thread.
+// source: https://man7.org/linux/man-pages/man3/pthread_detach.3.html
+int pthread_detach(pthread_t threadid)
+{
+__ESBMC_HIDE:;
+  __ESBMC_atomic_begin();
+  __ESBMC_assert(
+    !__ESBMC_pthread_thread_detach[(int)threadid],
```
Comment (rafaelsamenezes) [inline src/c2goto/library/pthread_lib.c:728]: Just add a minor comment about the EINVAL return is being checked here

File: src/c2goto/library/pthread_lib.c [hunk 2]
```diff
@@ -706,3 +711,26 @@ __ESBMC_HIDE:;
...
+    "Attempting to detach an already detached thread results in unspecified "
+    "behavior");
+  if(__ESBMC_pthread_thread_ended[(int)threadid])
```
Comment (rafaelsamenezes) [inline src/c2goto/library/pthread_lib.c]: Isn't this only checking if the thread has ended?

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]

File: src/c2goto/library/pthread_lib.c (hunk 1 above)
Comment (lucasccordeiro) [reply src/c2goto/library/pthread_lib.c:728]: Fixed.

File: src/c2goto/library/pthread_lib.c (hunk 2 above)
Comment (lucasccordeiro) [reply src/c2goto/library/pthread_lib.c]: Fixed.

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

**Pull request merged**